### PR TITLE
Added multiarch support by docker buildx

### PR DESCRIPTION
**Pull request by bivasda1**
Signed-off-by: Bivas Das <bivasda1@in.ibm.com>

### What does this PR do?
Enables docker buildx support in GitHub Actions job to build and publish multiarch docker images. All the workflows (nightly build, PR check, and release) are updated to have multiarch support.

### What issues does this PR fix or reference?

This refers to https://github.com/eclipse/che-devfile-registry/pull/321

### Reviewers
@ericwill Could you please review this PR

### CODE REVIEW

**Comment by bivasda1** [comment]
FYI - @nickboldt - Follow up PR as per comments here - https://issues.redhat.com/browse/CRW-1475?focusedCommentId=15721358&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15721358

File: .github/workflows/release.yml
Comment (ericwill) [inline .github/workflows/release.yml:2]: Copyright header needs to stay, but the year should be updated

File: .github/workflows/release.yml
Comment (ericwill) [inline .github/workflows/release.yml:41]: We need to log into both quay.io and dockerhub. The reason is that we push to quay, and we need to authenticate with dockerhub to prevent rate limiting.

File: .github/workflows/release.yml
Comment (ericwill) [inline .github/workflows/release.yml:26]: Any particular reason to use `v2` over `v1`?

File: .github/workflows/release.yml
Comment (ericwill) [inline .github/workflows/release.yml:56]: Both the happy path image and the base images need to be built with multi-arch support as well, otherwise the devfiles inside the registry will be no-op. I would suggest modifying the build scripts and adding `PLATFORMS` files.

File: .github/workflows/nightly-build-publish.yml
Comment (ericwill) [inline .github/workflows/nightly-build-publish.yml:59]: Both the happy path image and the base images need to be built with multi-arch support as well, otherwise the devfiles inside the registry will be no-op. I would suggest modifying the build scripts and adding `PLATFORMS` files.

**Review by ericwill — CHANGES_REQUESTED** [review]
A few changes are required, please see comments inline. Also, please test to make sure that the [regression](https://github.com/eclipse/che-devfile-registry/actions/runs/515108378) from the last attempt doesn't happen this time.

File: .github/workflows/release.yml
Comment (bivasda1) [reply .github/workflows/release.yml:26]: @ericwill no as such, Please let me know do you require v1 over v2?

Comment (ericwill) [reply .github/workflows/release.yml:26]: I don't think so, I see we are using `fetch-depth` so it shouldn't make a difference.

**Comment by bivasda1** [comment]
@ericwill Can you please review this PR, I have changed Accordingly
 
**Below change requests are updated**
- Copyright header and  also year is Updated
- Enabled log into both quay.io and dockerhub
- Enabled multiarch build using buildx for Both the happy path image and the base images 
- Added Platformfile

File: PLATFORMS
Comment (bivasda1) [inline PLATFORMS:1]: Platform file Added

File: arbitrary-users-patch/build_images.sh
Comment (bivasda1) [inline arbitrary-users-patch/build_images.sh:37]: Enabled docker buildx for multiarch build

File: .github/workflows/release.yml
Comment (bivasda1) [inline .github/workflows/release.yml:1]: copyright and year is updated

**Review by bivasda1 — COMMENTED** [review]
Change requests are updated

File: PLATFORMS
Comment (ericwill) [reply PLATFORMS:1]: I would prefer this file be added in `arbitrary-users-patch` and `arbitrary-users-patch/happy-path/`, and adjust the scripts accordingly. That way the build script can be run from the repository root, or from within the directories themselves.

**Comment by bivasda1** [comment]
@ericwill added Platform file in **arbitrary-users-patch and arbitrary-users-patch/happy-path** location, accordingly modified the script.

**Review by ericwill — DISMISSED** [review]

**Comment by ericwill** [comment]
@bivasda1 please rebase the PR and let's merge it next week. There is a `7.26.2` release next week so that will be a good test run.

**Comment by nickboldt** [comment]
@mkuznyetsov FYI ^

@bivasda1 if rebased and ready by EOD Tuesday, we can include in Wednesday's release. If not... will slip to the week after. 

Note too that we've been having issues with qemu emulated s390x buildx builds, so you might want to only include ppc64le in  this PR as the s390x build might make it harder to validate.

**Comment by bivasda1** [comment]
@nickboldt I will remove **s390x** form Platform files and will rebase as quickly as possible.

**Comment by bivasda1** [comment]
@nickboldt @ericwill I have rebased.

File: .github/workflows/release.yml
Comment (nickboldt) [inline .github/workflows/release.yml:80]: you're pushing the same image w/ 3 tags here:
*           quay.io/eclipse/${{ steps.prep.outputs.image }}:${{ steps.prep.outputs.version }},
*           quay.io/eclipse/${{ steps.prep.outputs.image }}:${{  github.event.inputs.version }}
*           quay.io/eclipse/${{ steps.prep.outputs.image }}:${{ steps.prep.outputs.short_sha1 }},

should it be only two pushes for prep.outputs.version and short_sha1? Not sure why we would want the submitted version 7.y.z AND version from VERSION file AND the short sha from the tip of the checked out branch. 

@mkuznyetsov WDYT?

Comment (bivasda1) [reply .github/workflows/release.yml:80]: @nickboldt 

> should it be only two pushes for prep.outputs.version and short_sha1?
Yes
```
prep.outputs.version-->7.27.0-SNAPSHOT
event.inputs.version-->7.27.0 
```
I was trying to release proper version, so that `event.inputs.version` I have added. Please let me know should I remove `event.inputs.version`?

I will remove spaces from `${{ github.event.inputs.version }}`

File: .github/workflows/release.yml
Comment (bivasda1) [inline .github/workflows/release.yml:80]: @nickboldt I have removed ` quay.io/eclipse/${{ steps.prep.outputs.image }}:${{  github.event.inputs.version }}` and rebased

File: .github/workflows/release.yml
Comment (nickboldt) [reply .github/workflows/release.yml]: we don't release -SNAPSHOT versions to quay.

https://quay.io/repository/eclipse/che-devfile-registry?tab=tags

Just nightly + sha tag and release (7.yy.z) + sha tag.

File: .github/workflows/release.yml
Comment (nickboldt) [reply .github/workflows/release.yml:80]: > outputs.version-->7.27.0-SNAPSHOT

That's the wrong version to release. We would want 7.27.0, not the -SNAPSHOT version. Releases are for releases, nightly is for snapshots. :D

File: .github/workflows/release.yml
Comment (nickboldt) [inline .github/workflows/release.yml:41]: If this value != `inputs.version` we should use inputs.version.

In fact I'm not sure why we need BOTH a computed value (7.27.0-SNAPSHOT) and an input value (7.27.0). 

I would expect that if the input is missing, we compute it... but I believe part of the release process is to SET the value in the VERSION file, so there's no point reading the file before we set the value. :)  

Instead I'd just take the value from the input as it should be the same as the VERSION file, once the release is complete. 

Correct @mkuznyetsov ?

Comment (bivasda1) [reply .github/workflows/release.yml:41]: @nickboldt do you expect ci-multiarch workflows should be formatted https://github.com/eclipse/che-dashboard/blob/master/.github/workflows/ci-multiarch.yaml#L91-L115 referenced to https://github.com/eclipse/che-dashboard/pull/119#issuecomment-783374087

**Comment by nickboldt** [comment]
> do you expect ci-multiarch workflows should be formatted

Since I don't own this project, I'll let the TL comment... but my $0.02 is that it seems like from a project management perspective it might be better/more efficient to have:

* 1 GH action that can handle all three (four) arches, rather than 
* 1 GH action action for x64 and ppc64 (and aarch64), while a travis script separately handles s390x.

File: .github/workflows/release.yml
Comment (mkuznyetsov) [inline .github/workflows/release.yml:18]: I'd suggest having an empty default value, so we wouldn't have accidental workflow runs with `7.y.z` taken as a default parameter.

File: .github/workflows/release.yml
Comment (bivasda1) [reply .github/workflows/release.yml:41]: > Instead I'd just take the value from the input as it should be the same as the VERSION file, once the release is complete.

@nickboldt can we modify the version file `7.27.0-SNAPSHOT->7.27.0` apply the condition if `[vesion==input.version]` then release the specific version

Comment (mkuznyetsov) [reply .github/workflows/release.yml:41]: @nickboldt as of now, make-release.sh is called first from another workflow, which makes changes to VERSION file, and then this workflow does the rest. Here it reads the VERSION file, in case it was triggered by the push to the release branch, which has no inputs

Because it was slightly confusing, I made a PR (about which I've forgot 😃 )
https://github.com/eclipse/che-devfile-registry/pull/338/files
it makes `make-release.sh` will be called from this workflow

**Comment by bivasda1** [comment]
> > do you expect ci-multiarch workflows should be formatted https://github.com/eclipse/che-dashboard/blob/master/.github/workflows/ci-multiarch.yaml#L91-L115 referenced to eclipse/che-dashboard#119
> 
> Since I don't own this project, I'll let the TL comment... but my $0.02 is that it seems like from a project management perspective it might be better/more efficient to have:
> 
> * 1 GH action that can handle all three (four) arches, rather than
> * 1 GH action action for x64 and ppc64 (and aarch64), while a travis script separately handles s390x.



> > do you expect ci-multiarch workflows should be formatted
> 
> Since I don't own this project, I'll let the TL comment... but my $0.02 is that it seems like from a project management perspective it might be better/more efficient to have:
> 
> * 1 GH action that can handle all three (four) arches, rather than
> * 1 GH action action for x64 and ppc64 (and aarch64), while a travis script separately handles s390x.

@ericwill Could you please share your thoughts @nickboldt suggestions

**Comment by bivasda1** [comment]
@nickboldt Could you please review this PR,I think Visioning issue is resolved  now.

**Review by ericwill — COMMENTED** [review]
Please remove this file: `.github/workflows/.nightly-build-publish.yml.swp`

**Comment by ericwill** [comment]
> > > do you expect ci-multiarch workflows should be formatted https://github.com/eclipse/che-dashboard/blob/master/.github/workflows/ci-multiarch.yaml#L91-L115 referenced to [eclipse/che-dashboard#119](https://github.com/eclipse/che-dashboard/pull/119)
> > 
> > 
> > Since I don't own this project, I'll let the TL comment... but my $0.02 is that it seems like from a project management perspective it might be better/more efficient to have:
> > 
> > * 1 GH action that can handle all three (four) arches
> >   rather than
> > * 1 GH action action for x64 and ppc64 (and aarch64), while a travis script separately handles s390x.
> 
> @ericwill Could you please share your thoughts @nickboldt suggestions

I agree with the first option: `1 GH action that can handle all three (four) arches`

**Comment by bivasda1** [comment]
> Please remove this file: `.github/workflows/.nightly-build-publish.yml.swp`

Opps,I will remove that.Thanks

**Comment by bivasda1** [comment]
> > Please remove this file: `.github/workflows/.nightly-build-publish.yml.swp`
> 
Could u please review the PR I have  deleted `.github/workflows/.nightly-build-publish.yml.swp
`

File: .github/workflows/release.yml
Comment (ericwill) [reply .github/workflows/release.yml:41]: So which approach are we taking? Right now as it stands, the `7.27.0-SNAPSHOT` problem won't happen because the workflow is triggered after the file is changed. But we can have this PR wait until https://github.com/eclipse/che-devfile-registry/pull/338/files is merged. What do you guys think? cc @nickboldt @mkuznyetsov

Comment (nickboldt) [reply .github/workflows/release.yml:41]: PR 338 is ready to merge, after fixing all the shellcheck complaints. https://github.com/eclipse/che-devfile-registry/pull/338/files

Comment (ericwill) [reply .github/workflows/release.yml:41]: I've merged it, @bivasda1 please adjust the PR to make sure `-SNAPSHOT` isn't pushed as part of the tag

**Comment by nickboldt** [comment]
please rebase, you're out of date again :(

**Comment by bivasda1** [comment]
@nickboldt Rebased and Added buildx support `make-release.sh`

**Review by ericwill — APPROVED** [review]
Let's give it a shot

**Comment by nickboldt** [comment]
seems legit for x and power arches but this won't work for s390x and will need to be refactored in future.

+1 for pushing this for now, with the caveat that we'll need something like https://github.com/eclipse/che-dashboard/pull/177 to support pushing each per-arch to quay independently, then combining them into a single multi-arch digest/manifest.

**Review by nickboldt — APPROVED** [review]
+1, but won't work for s390x.

**Comment by ArvinB** [comment]
Just commenting...here is a bash script very similar to what I do for building multi-arch images. As @nickboldt mentioned from https://github.com/eclipse/che-dashboard/pull/177 that there is a risk of building for `s390x`. My issue primarily is that the CentOS repo's do not contain the packages for `s390x`. I don't have enough knowledge yet on which repo(s) to include, so for now I disable them all so that basically the yum and dnf updates all just pass on through.

```
registry=quay.io
repo=some/path
version=1.0.0
platforms=("amd64" "s390x")

tags=""
for platform in ${platforms[*]}
do
  echo "Building for linux/$platform"
  tag=$registry/$repo:${version}_$platform
  docker buildx build --load --platform linux/$platform -t $tag .
  docker push $tag
  tags="${tags} ${tag}"
done

docker manifest create $registry/$repo:$version $tags
docker manifest push $registry/$repo:$version
docker buildx imagetools inspect $registry/$repo:$version
```

FYI
@sleshchenko There is great interest in building multi-arch.

**Comment by ericwill** [comment]
@bivasda1 so the build was successful, but please fix/update the following in a new PR:
* the `7.28.0-SNAPSHOT` tag shouldn't be there
* a nightly tag hasn't been published/updated -- likely this is what the `7.28.0-SNAPSHOT` tag should be

https://quay.io/repository/eclipse/che-devfile-registry?tab=tags

**Comment by nickboldt** [comment]
There, I fixed it. :) https://quay.io/repository/eclipse/che-devfile-registry?tab=tags&tag=nightly

Also deleted the 7.28.0-SNAPSHOT tag

**Comment by nickboldt** [comment]
But also... the image is not multiarch :(

```
#23 [linux/arm64 5/6] RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh
#23 sha256:3075bc83c5d59c32a9b1a25f5afd028722d2f71f259502bc813ddbcec8d90bf5
#23 0.092 /bin/sh: /tmp/install-editor-tooling.sh: not found
#23 ERROR: executor failed running [/bin/sh -c /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh]: exit code: 127

#12 [linux/amd64 5/6] RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh
#12 sha256:feb2a9698386ef81638fdeab39c0d81e0bbdad912bc578d7027b6a30d8144c40
#12 0.229 container process is already dead
#12 CANCELED

#17 [linux/ppc64le 5/6] RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh
#17 sha256:6d32b5832d93ea6889e549b5179b2c324f77d75fb4ce4be71c5176c98682b25b
#17 0.235 container process is already dead
#17 CANCELED
------
 > [linux/arm64 5/6] RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh:
------
Dockerfile:10
--------------------
   8 |     # Install common terminal editors in container to aid development process
   9 |     COPY install-editor-tooling.sh /tmp
  10 | >>> RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh
  11 |     
  12 |     USER 10001
--------------------
error: failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh]: exit code: 127
Pushing quay.io/eclipse/che-antora-2.3:nightly to remote registry
An image does not exist locally with the tag: quay.io/eclipse/che-antora-2.3
The push refers to repository [quay.io/eclipse/che-antora-2.3]
Building quay.io/eclipse/che-cpp-rhel7:nightly based on registry.access.redhat.com/devtools/llvm-toolset-rhel7 ...
time="2021-03-08T15:15:42Z" level=warning msg="No output specified for docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load"
```

-- https://github.com/eclipse/che-devfile-registry/runs/2057571221?check_suite_focus=true

and if I skopeo inspect I only get amd64:

```
$➔ skopeo --override-os ppc64le inspect docker://quay.io/eclipse/che-devfile-registry:nightly | jq -r .Architecture
amd64
```

Contrast with:

```
$➔ skopeo inspect docker://quay.io/eclipse/che-dashboard:multiarch-next --raw | jq -r '.manifests[].platform.architecture'
amd64
arm64
ppc64le
s390x
```

**Comment by ArvinB** [comment]
@nickboldt I curious how this will get resolved, because this is similar to the issue I faced when trying to build registry images for `s390x`.

**Comment by nickboldt** [comment]
well, there's the approach in https://github.com/eclipse/che-dashboard/blob/master/.github/workflows/ci-multiarch.yaml ...

File: .github/workflows/nightly-build-publish.yml
Comment (Prabhav-Thali) [inline .github/workflows/nightly-build-publish.yml:70]: @nickboldt Image is still single arch because I guess slight change is required here. Instead of `platforms: ${{ steps.package.outputs.content }}`, it needs to be `platforms: ${{ steps.prep.outputs.platforms }}` as defined on line 37.

**Comment by bivasda1** [comment]
> But also... the image is not multiarch :(
> 
> ```
> #23 [linux/arm64 5/6] RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh
> #23 sha256:3075bc83c5d59c32a9b1a25f5afd028722d2f71f259502bc813ddbcec8d90bf5
> #23 0.092 /bin/sh: /tmp/install-editor-tooling.sh: not found
> #23 ERROR: executor failed running [/bin/sh -c /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh]: exit code: 127
> 
> #12 [linux/amd64 5/6] RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh
> #12 sha256:feb2a9698386ef81638fdeab39c0d81e0bbdad912bc578d7027b6a30d8144c40
> #12 0.229 container process is already dead
> #12 CANCELED
> 
> #17 [linux/ppc64le 5/6] RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh
> #17 sha256:6d32b5832d93ea6889e549b5179b2c324f77d75fb4ce4be71c5176c98682b25b
> #17 0.235 container process is already dead
> #17 CANCELED
> ------
>  > [linux/arm64 5/6] RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh:
> ------
> Dockerfile:10
> --------------------
>    8 |     # Install common terminal editors in container to aid development process
>    9 |     COPY install-editor-tooling.sh /tmp
>   10 | >>> RUN /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh
>   11 |     
>   12 |     USER 10001
> --------------------
> error: failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c /tmp/install-editor-tooling.sh && rm -f /tmp/install-editor-tooling.sh]: exit code: 127
> Pushing quay.io/eclipse/che-antora-2.3:nightly to remote registry
> An image does not exist locally with the tag: quay.io/eclipse/che-antora-2.3
> The push refers to repository [quay.io/eclipse/che-antora-2.3]
> Building quay.io/eclipse/che-cpp-rhel7:nightly based on registry.access.redhat.com/devtools/llvm-toolset-rhel7 ...
> time="2021-03-08T15:15:42Z" level=warning msg="No output specified for docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load"
> ```
> 
> -- https://github.com/eclipse/che-devfile-registry/runs/2057571221?check_suite_focus=true
> 
> and if I skopeo inspect I only get amd64:
> 
> ```
> $➔ skopeo --override-os ppc64le inspect docker://quay.io/eclipse/che-devfile-registry:nightly | jq -r .Architecture
> amd64
> ```
> 
> Contrast with:
> 
> ```
> $➔ skopeo inspect docker://quay.io/eclipse/che-dashboard:multiarch-next --raw | jq -r '.manifests[].platform.architecture'
> amd64
> arm64
> ppc64le
> s390x
> ```
@nickboldt @ericwill 
there are these images which are not multiarch. So do we need to excluded from build for releasing multiarch? 
```
che-antora-2.3          docker.io/antora/antora:2.3.3
che-dotnet-2.2          mcr.microsoft.com/dotnet/core/sdk:2.2-stretch
che-dotnet-3.1          mcr.microsoft.com/dotnet/core/sdk:3.1.301-buster
che-nodejs8-centos      registry.centos.org/che-stacks/centos-nodejs
che-php-7               quay.io/eclipse/che-php-base:7.4
che-python-3.6          centos/python-36-centos7:1

che-cpp-rhel7           registry.access.redhat.com/devtools/llvm-toolset-rhel7->not compatible for arm64
che-rust-1.39           rust:1.39.0-slim->not compatible for ppc64le
```

**Comment by ericwill** [comment]
> @nickboldt @ericwill
> there are these images which are not multiarch. So do we need to excluded from build for releasing multiarch?
> 
> ```
> che-antora-2.3          docker.io/antora/antora:2.3.3
> che-dotnet-2.2          mcr.microsoft.com/dotnet/core/sdk:2.2-stretch
> che-dotnet-3.1          mcr.microsoft.com/dotnet/core/sdk:3.1.301-buster
> che-nodejs8-centos      registry.centos.org/che-stacks/centos-nodejs
> che-php-7               quay.io/eclipse/che-php-base:7.4
> che-python-3.6          centos/python-36-centos7:1
> 
> che-cpp-rhel7           registry.access.redhat.com/devtools/llvm-toolset-rhel7->not compatible for arm64
> che-rust-1.39           rust:1.39.0-slim->not compatible for ppc64le
> ```
> 
> Even we can enable multiarch for these repos.
> 
> ```
> che-php-7               quay.io/eclipse/che-php-base:7.4->registry.access.redhat.com/ubi8/php-72
> che-python-3.6          centos/python-36-centos7:1->registry.access.redhat.com/ubi8/python-36
> ```

Yes we need to exclude stacks that aren't available on multi-arch. For example if rust is x86 only, then it should be excluded from builds on other arches.

**Comment by nickboldt** [comment]
For arch specific devfile exclusion filtering, see what we're dong downstream in CRW:

* if there's a .s390x or .ppc64le version of a devfile, use that instead of the default. If file is empty, remove the devfile entirely
  * https://github.com/redhat-developer/codeready-workspaces/blob/crw-2.7-rhel-8/dependencies/che-devfile-registry/build/scripts/swap_yamlfiles.sh
  * https://github.com/redhat-developer/codeready-workspaces/tree/crw-2.6-rhel-8/dependencies/che-devfile-registry/devfiles/00_java-microprofile-xp2 (x only)
  * https://github.com/redhat-developer/codeready-workspaces/tree/crw-2.6-rhel-8/dependencies/che-devfile-registry/devfiles/00_java-microprofile-xp1 (z & p only)

* switch from openjdk to openj9 image:
  *  https://github.com/redhat-developer/codeready-workspaces/blob/crw-2.7-rhel-8/dependencies/che-devfile-registry/build/scripts/swap_images.sh

* if the devfile references a sidecar which doesn't exist (or doesn't exist for the current arch), remove the devfile while adding digests:
  * https://github.com/redhat-developer/codeready-workspaces/blob/crw-2.7-rhel-8/dependencies/che-devfile-registry/build/scripts/write_image_digests.sh#L22-L26

**Comment by bivasda1** [comment]
> For arch specific devfile exclusion filtering, see what we're dong downstream in CRW:
> 
> * if there's a .s390x or .ppc64le version of a devfile, use that instead of the default. If file is empty, remove the devfile entirely
>   
>   * https://github.com/redhat-developer/codeready-workspaces/blob/crw-2.7-rhel-8/dependencies/che-devfile-registry/build/scripts/swap_yamlfiles.sh
>   * https://github.com/redhat-developer/codeready-workspaces/tree/crw-2.6-rhel-8/dependencies/che-devfile-registry/devfiles/00_java-microprofile-xp2 (x only)
>   * https://github.com/redhat-developer/codeready-workspaces/tree/crw-2.6-rhel-8/dependencies/che-devfile-registry/devfiles/00_java-microprofile-xp1 (z & p only)
> * switch from openjdk to openj9 image:
>   
>   * https://github.com/redhat-developer/codeready-workspaces/blob/crw-2.7-rhel-8/dependencies/che-devfile-registry/build/scripts/swap_images.sh
> * if the devfile references a sidecar which doesn't exist (or doesn't exist for the current arch), remove the devfile while adding digests:
>   
>   * https://github.com/redhat-developer/codeready-workspaces/blob/crw-2.7-rhel-8/dependencies/che-devfile-registry/build/scripts/write_image_digests.sh#L22-L26

@nickboldt Thanks for your details :)

**Pull request merged**
